### PR TITLE
Refactor embedUrl to create base URL from origin

### LINKED ISSUE #3133: AV: Embed URL should use current origin as base URL
**Is your feature request related to a problem? Please describe.**
Instead of varying the base embed URL based on the environment, it should use the base URL of the article currently being viewed, e.g.
- `https://www.test.bbc.co.uk/news/articles/c3wmq4d1y3wo` will have an embed base URL of `https://www.test.bbc.co.uk`
- `https://www.test.bbc.com/news/articles/c3wmq4d1y3wo` will have an embed base URL of `https://www.test.bbc.com`
- `https://www.bbc.com/news/articles/foo` will have an embed base URL of `https://www.bbc.com`

This will help reduce problems caused by framing content served over different domains.

**Describe the solution you'd like**
Modify the `embedUrl` function to use the current base URL as the base URL for the embed URL:
https://github.com/bbc/simorgh/blob/fee8262b367787d35d68fd3244888ffd4db639f2/src/app/containers/MediaPlayer/helpers/embedUrl.js#L1-L10

**Testing notes**
[Tester to complete]

Dev insight: Snapshot and unit tests should suffice.

**Additional context**
We will still be doing bbc/simorgh-infrastructure#641. Both this issue and the other will fix problems serving media player on .com and .co.uk`.

### PR DESCRIPTION
**Pull request by 12**
Resolves #3133 

**Overall change:** Instead of varying the base URL for AV embeds based on what's defined in `envConfig` files, we should vary it based on the `origin` set in `RequestContext`.

**Code changes:**

- New `buildBaseUrl` method that will build an embed URL based on the `origin` it is passed.
- Pass `origin` from `RequestContext` to `embedUrl` helper
- Better testing of `embedUrl` helper function

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/MediaPlayer/helpers/embedUrl.js [hunk 1]
```diff
@@ -1,6 +1,18 @@
-const embedUrl = (assetId, vpid, amp = false) => {
-  const baseUrl = process.env.SIMORGH_AV_EMBED_BASE_URL;
-  const parts = [baseUrl, assetId, vpid];
+const buildBaseUrl = origin => {
+  if (!origin.includes('local')) {
+    return origin;
+  }
+
+  const tld = origin.includes('.co.uk') ? '.co.uk' : '.com';
+  const baseUrl = `https://www.test.bbc${tld}`;
```
Comment (dr3) [inline src/app/containers/MediaPlayer/helpers/embedUrl.js]: Just return this, doesn't need to be made into a const :)

File: src/app/containers/MediaPlayer/index.jsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ const MediaPlayerContainer = ({ blocks }) => {
     return null; // this should be the holding image with an error overlay
   }
 
-  const embedSource = embedUrl(id, versionId, isAmp);
+  const embedSource = embedUrl(origin, id, versionId, isAmp);
```
Comment (dr3) [inline src/app/containers/MediaPlayer/index.jsx]: Can you make `embedUrl` take an object as its argument plz

**Review by dr3 — CHANGES_REQUESTED** [review]
lgtm, just a couple things

File: src/app/containers/MediaPlayer/index.jsx [hunk 3]
```diff
@@ -70,9 +70,14 @@ const MediaPlayerContainer = ({ blocks }) => {
...
   }
 
-  const embedSource = embedUrl(id, versionId, isAmp);
-  const placeholderSrc = placeholderImage(imageUrl);
   const Player = isAmp ? AmpMediaPlayer : CanonicalMediaPlayer;
+  const placeholderSrc = placeholderImage(imageUrl);
+  const embedSource = embedUrl({
+    vpid: versionId,
+    assetId: id,
+    amp: isAmp,
```
Comment (dr3) [inline src/app/containers/MediaPlayer/index.jsx]: Can we keep this as `isAmp` not `amp` please, after that it has my 👍

**Review by dr3 — APPROVED** [review]

**Review by rhenshaw56 — APPROVED** [review]

**Pull request merged**